eclipse/che
# Setup Sonar check for master and Github pull requests

### What does this PR do?
Setup Sonar check for master and Github pull requests. Results can be found here `https://sonarcloud.io/dashboard?id=org.eclipse.che%3Ache-server`
I`ve decided not to change che-parent. But first, get stable sonar testing results, and after that with @nickboldt @mkuznyetsov can decide whatever we should move the sonar config to https://github.com/eclipse/che-parent or not.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18507

### How to test this PR?
- Setup `SONAR_TOKEN`
-  `mvn -B clean verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Psonar -Pintegration`


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
